File: Design Patterns/Facade Pattern/Python/LojaFacade.py

from ProcessadorPedidos import ProcessadorPedidos
from SistemaEnvio import SistemaEnvios
from SistemaPagamentos import SistemaPagamentos



class LojaFacade:
    def __init__(self):
        self.processador_pedidos = ProcessadorPedidos()
        self.sistema_pagamento = SistemaPagamentos()
        self.sistema_envio = SistemaEnvios()

    def fazer_pedido(self, item, valor):
        self.processador_pedidos.criar_pedido(item)
        self.sistema_pagamento.processar_pagamento(valor)
        self.sistema_envio.enviar_pedido(item)

        print("Pedido realizado com sucesso!")
        

File: Design Patterns/Facade Pattern/Java/com.faculdadevincit.desingpatterns.facadepattern/src/LojaFacade.java
public class LojaFacade {
    private ProcessadorPedidos processadorPedidos;
    private SistemaPagamentos sistemaPagamentos;
    private SistemEnvio sistemaEnvio;

    public LojaFacade() {
        processadorPedidos = new ProcessadorPedidos();
        sistemaEnvio = new SistemEnvio();
        sistemaPagamentos = new SistemaPagamentos();
    }

        public void fazerPedido(String item, double valor) {
            processadorPedidos.CriarPedido(item);
            sistemaPagamentos.ProcessarPagamento(valor);
            sistemaEnvio.EnviarPedido(item);

            System.out.println("Pedido realizado com sucesso");
        }

}
